PSPs/psp-17.md

```diff
@@ -36,51 +36,113 @@ The main motivation for this proposal is to have one **trait** that shares the s
 as naming of trait affects the identifiers of functions in this trait.
 The second motivation is to define an exhaustive method list in this trait. Unlike ERC20, we suggest including `increase_allowance` & `decrease_allowance`
 as a part of standard proposal and extract metadata fields to separate trait.
-
-
-### Types
-```rust
-// AccountId is 32 bytes array, like in substrate-based blockchains.
-type AccountId = [u8; 32];
-// u128 must be enough to cover most of the use cases of standard token.
-type Balance = u128;
-```
+Another suggestion is to have `PSP17Receiver` and call `on_received` method at the end of transfer if the recipient is a contract.
 
 ### Traits
 
 ```rust
+/// PSP17 is a trait of Fungible Token Standard
 pub trait PSP17 {
+ /// Returns the total token supply.
  fn total_supply(&self) -> Balance;
 
+ /// Returns the account balance for the specified `owner`.
+ /// Returns `0` if the account is non-existent.
  fn balance_of(&self, owner: AccountId) -> Balance;
 
- fn transfer(&mut self, to: AccountId, value: Balance);
-
+ /// Returns the amount which `spender` is still allowed to withdraw from `owner`.
+ /// Returns `0` if no allowance has been set.
  fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance;
 
- fn transfer_from(&mut self, from: AccountId, to: AccountId, value: Balance);
-
+ /// Transfers `value` amount of tokens from the caller's account to account `to`
+ /// with additional `data` with no specified format.
+ /// Emits a `Transfer` event on success.
+ /// 
+ /// This method also calls [on_received](#on_received) method on `to`.
+ /// 
+ /// **Errors**
+ /// * Panics with `InsufficientBalance` error if there are not enough tokens on
+ /// the caller's account Balance.
```

Comment:
balance should be lowercased, it refers to literal balance, not the type in ink!.